src/gatling/simulations/scenarios/Solicitor_Bails.scala

```diff
@@ -0,0 +1,1383 @@
+          .header("accept", "application/json")
+          .check(substring("Immigration & Asylum")))
+
+    }
+    .pause(MinThinkTime, MaxThinkTime)
+
+    /*=============================================================================================
+    * Jurisdiction = Immigration & Asylum; Case Type = Bail* master; Event = Start The Application
+    ===============================================================================================*/
+
+    .group("XUI_Bails_040_SelectCaseType") {
+      exec(Common.healthcheck("%2Fcases%2Fcase-create%2FIA%2FBail%2FstartApplication"))
+
+      .exec(http("XUI_Bails_040_005_SelectCaseType")
+        .get(BaseURL + "/data/internal/case-types/Bail/event-triggers/startApplication?ignore-warning=false")
+        .headers(Headers.commonHeader)
+        .header("accept", "application/vnd.uk.gov.hmcts.ccd-data-store-api.ui-start-case-trigger.v2+json;charset=UTF-8")
+        .check(jsonPath("$.event_token").saveAs("event_token"))
+        .check(jsonPath("$.id").is("startApplication")))
+
+      .exec(Common.healthcheck("%2Fcases%2Fcase-create%2FIA%2FBail%2FstartApplication%2FstartApplicationhasPreviousBailApplication"))
+
+      .exec(getCookieValue(CookieKey("XSRF-TOKEN").withDomain(BaseURL.replace("https://", "")).saveAs("XSRFToken")))
+    }
+    .pause(MinThinkTime, MaxThinkTime)
+
+    /*======================================================================================
+    * Previous applications? - No
+    ======================================================================================*/
+
+    .group("XUI_Bails_050_Previous_Applications") {
+      exec(http("XUI_Bails_050_005_Previous_Applications")
+        .post(BaseURL + "/data/case-types/Bail/validate?pageId=startApplicationhasPreviousBailApplication")
+        .headers(Headers.commonHeader)
+        .header("accept", "application/vnd.uk.gov.hmcts.ccd-data-store-api.case-data-validate.v2+json;charset=UTF-8")
+        .header("x-xsrf-token", "${XSRFToken}")
+        .body(ElFileBody("bodies/bails/BailsPreviousApplications.json"))
+        .check(substring("hasPreviousBailApplication")))
+
+    }
+    .pause(MinThinkTime, MaxThinkTime)
+
+    /*======================================================================================
+    * Create new application
+    ======================================================================================*/
+
+    .group("XUI_Bails_060_CreateNewApplication") {
+      exec(http("XUI_Bails_060_005_CreateNewApplication")
+        .post(BaseURL + "/data/case-types/Bail/validate?pageId=startApplicationnoAccessToPreviousApplications")
+        .headers(Headers.commonHeader)
+        .header("accept", "application/vnd.uk.gov.hmcts.ccd-data-store-api.case-data-validate.v2+json;charset=UTF-8")
+        .header("x-xsrf-token", "${XSRFToken}")
+        .body(ElFileBody("bodies/bails/BailsCreateNewApplication.json"))
+        .check(regex("startApplicationnoAccessToPreviousApplications")))
```

Comment:
All these regexes for a single string should be replaced with a substring function, as they are lot more efficient - unless you need to use a regex.

Reply:
Used a regex because there's nothing to search up in the 'preview section' for a substring search